lib/components/user/trip-basics-pane.js

```diff
@@ -31,73 +37,134 @@ const StyledLabel = styled.label`
 `
 
 const allDays = [
-  { name: 'monday', text: 'Mon.' },
-  { name: 'tuesday', text: 'Tue.' },
-  { name: 'wednesday', text: 'Wed.' },
-  { name: 'thursday', text: 'Thu.' },
-  { name: 'friday', text: 'Fri.' },
-  { name: 'saturday', text: 'Sat.' },
-  { name: 'sunday', text: 'Sun.' }
+  { name: 'monday', text: 'Mon.', fullText: 'Mondays' },
+  { name: 'tuesday', text: 'Tue.', fullText: 'Tuesdays' },
+  { name: 'wednesday', text: 'Wed.', fullText: 'Wednesdays' },
+  { name: 'thursday', text: 'Thu.', fullText: 'Thursdays' },
+  { name: 'friday', text: 'Fri.', fullText: 'Fridays' },
+  { name: 'saturday', text: 'Sat.', fullText: 'Saturdays' },
+  { name: 'sunday', text: 'Sun.', fullText: 'Sundays' }
 ]
 
 /**
  * This component shows summary information for a trip
  * and lets the user edit the trip name and day.
  */
-const TripBasicsPane = ({ errors, touched, values: monitoredTrip }) => {
-  const { itinerary } = monitoredTrip
-
-  if (!itinerary) {
-    return <div>No itinerary to display.</div>
-  } else {
-    // Show an error indicaton when monitoredTrip.tripName is not blank (from the form's validation schema)
-    // and that tripName is not already used.
-    let tripNameValidationState = null
-    if (touched.tripName) {
-      tripNameValidationState = errors.tripName ? 'error' : null
-    }
+class TripBasicsPane extends Component {
+  constructor () {
+    super()
+
+    this.state = {}
+  }
+
+  async componentDidMount () {
+    // Check itinerary existence for all days.
+    const { accessToken, config, setFieldValue, values } = this.props
+
+    const itineraryCheckResult = await checkItinerary(
+      config.persistence.otp_middleware,
+      accessToken,
+      values
+    )
```

Comment:
Since this check occurs every time the component mounts, it might be possible that an itinerary becomes impossible for an existing itinerary that was possible before. Is there a way to have this run only when creating a monitored trip?

Reply:
That was my initial thought and It is possible from the UI, but it will currently fail on the backend (the `preUpdateHook` method will trigger an error if the trip does not exist for one of the monitored days). Should the backend be changed to accommodate that?

Reply:
We should probably think about this a little more in depth. At some point this UI should also show when an itinerary is no longer possible which means updating things in the backend and then communicating that somehow to the frontend.

Reply:
I went ahead and remove the trip existence enforcement when saving existing trips in https://github.com/ibi-group/otp-middleware/pull/77/commits/78e31e7b50bcda047f65ddf1cbbbe506dbb8e455. The UI will still perform the check and highlight the days an existing trip is not possible, however it will let the user save that trip. Trip existence continues to be enforced for new trips.